pkg/kubelet/prober/prober.go

```diff
@@ -231,11 +231,7 @@ func (p *prober) newExecInContainer(container api.Container, containerID kubecon
 		var buffer bytes.Buffer
 		output := ioutils.WriteCloserWrapper(&buffer)
 		err := p.runner.ExecInContainer(containerID, cmd, nil, output, output, false, nil)
-		if err != nil {
-			return nil, err
-		}
-
-		return buffer.Bytes(), nil
+		return buffer.Bytes(), err
```

Comment:
Will err be an ExitError in the case this PR wants to fix? It's unidiomatic to return actual values if `err != nil`. A more explicit interface, e.g. with an additional `rc int` would be better IMO.